assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentServiceImpl.java

```diff
@@ -1403,32 +1403,58 @@ public void updateAssignment(Assignment assignment) throws PermissionException {
-                        authzGroupService.save(group);
-                    } catch (GroupNotDefinedException | AuthzPermissionException e) {
-                        log.warn("Exception while removing lock for assignment {}, {}", assignment.getId(), e.toString());
+                Collection<String> currentAssignedGroupRefs = assignment.getGroups();
+                // get the unassigned Group References by removing the current ones !
+                prevAssignedGroupRefs.removeAll(currentAssignedGroupRefs);
+
+                Site site = null;
+                try {
+                    site = siteService.getSite(assignment.getContext());
+                } catch (IdUnusedException e) {
+                    log.warn("Could not get the site {} for assignment {} while updating it", assignment.getContext(), assignment.getId());
+                }
+                // on Lessons, a group assignment or one with prereq. will be managed through a single (access) group
+                Group anAssignedGroup = site.getGroup(currentAssignedGroupRefs.stream().findFirst().orElse(null));
+                // we can infer it is an ACCESS group if it has the following property
+                String assignedGroupLessonRef = anAssignedGroup.getProperties().getProperty("lessonbuilder_ref");
+
+                // If the assigned group IS NOT an ACCESS group...
+                if (assignedGroupLessonRef == null) {
+                    for (String groupRef : prevAssignedGroupRefs) { // remove locks for groups that were removed or unassigned
+                        try {
+                            AuthzGroup group = authzGroupService.getAuthzGroup(groupRef);
+                            group.setLockForReference(reference, AuthzGroup.RealmLockMode.NONE);
+                            authzGroupService.save(group);
+                        } catch (GroupNotDefinedException | AuthzPermissionException e) {
+                            log.warn("Exception while removing lock for assignment {}, {}", assignment.getId(), e.toString());
+                        }
                     }
                 }
                 if (!assignment.getDraft()) { // don't add locks for draft assignments
                     if (assignment.getIsGroup()) { // lock mode ALL for group assignments
-                        for (String groupRef : assignment.getGroups()) {
+                        for (String groupRef : currentAssignedGroupRefs) {
                             try {
+                                Group assignedGroup = site.getGroup(groupRef);
+                                String lessonbuilderRef = assignedGroup.getProperties().getProperty("lessonbuilder_ref");
+
                                 AuthzGroup group = authzGroupService.getAuthzGroup(groupRef);
-                                group.setLockForReference(reference, AuthzGroup.RealmLockMode.ALL);
+                                if (lessonbuilderRef == null) {
```

Comment:
Let's be simple:

// exception: it's an ACCESS group, membership won't be locked
// as users are dynamically added to be able to see the related assignment on Lessons
group.setLockForReference(reference, lessonbuilderRef == null ? AuthzGroup.RealmLockMode.ALL : AuthzGroup.RealmLockMode.DELETE);